azurelinuxagent/ga/exthandlers.py

```diff
@@ -783,6 +783,19 @@ def __init__(self, ext_handler, protocol):
             self.logger.error(u"Failed to create extension log dir: {0}", e)
 
         log_file = os.path.join(self.get_log_dir(), "CommandExecution.log")
+
+        try:
+            if os.stat(log_file).st_size > execution_log_max_size:
+                with open(log_file, "rb") as existing_log_file_buffer:
+                    existing_log_file_buffer.seek(-1 * execution_log_max_size, 2)
+                    _ = existing_log_file_buffer.readline()
```

Comment:
readline won't have issues handling a binary file?

Reply:
No, as far as I can tell (via testing and posts), it is even utf-8 aware.

Reply:
ok